weave/trace/op.py

```diff
@@ -359,13 +387,14 @@ def _call_sync_func(
     # returns.  The user must explicitly call `finish` on the call object.  This is
     # included to support the imperative evaluation logging interface.
     __require_explicit_finish: bool = False,
+    client: ClientInterface | None = None,
     **kwargs: Any,
 ) -> tuple[Any, Call]:
     func = op.resolve_fn
     call = placeholder_call()
 
     # Handle all of the possible cases where we would skip tracing.
-    if is_tracing_setting_disabled() or should_skip_tracing_for_op(op):
+    if False:  # is_tracing_setting_disabled() or should_skip_tracing_for_op(op):
```

Comment:
This is still an outstanding problem. Unsure how to solve this.
The global context has no `WeaveClient` so the code things that tracing is disabled. I cannot use `set_weave_client_global` as each incoming call has potentially a different `project_id` so the client needs to be set per-call. And since they are processed asynchronously, using `set_weave_client_global` leads to race conditions and thread-unsafety.